Sources/StreamChat/Utils/Database/NSManagedObject+Extensions.swift

```diff
@@ -16,7 +16,7 @@ extension NSEntityDescription {
         for request: NSFetchRequest<T>
     ) -> T {
         let entity = insertNewObject(forEntityName: T.entityName, into: context)
-        request.entity = NSEntityDescription.entity(forEntityName: T.entityName, in: context)!
+        request.entity = T(context: context).entity
```

Comment:
This will create a new object and insert it into the specified context so you'll want to replace `let entity = insertNewObject(forEntityName: T.entityName, into: context)` with `let entity = T(context: context)` instead. And then use `request.entity = entity.entity` (might want to rename entity)

Reply:
fixed!